### PR TITLE
feat(bitbucket): Enable bitbucket reviewers

### PR DESCRIPTION
**Pull request by mdlavin**
This adds support for adding reviewers to Bitbucket PRs.

I've tested that this works against our Bitbucket repositories and it works for me.

Thank you so much for adding Bitbucket support!

### CODE REVIEW

**Comment by mdlavin** [comment]
Opps, I didn't use consistent naming in my commit summaries. I'll update the PR with better summaries in 30 min

File: lib/platform/bitbucket/index.js [hunk 1]
```diff
@@ -385,10 +385,19 @@ function addAssignees() {
...
-  return Promise.resolve();
+async function addReviewers(prId, reviewers) {
+  logger.debug(`Adding reviewers ${reviewers} to #${prId}`);
+
+  const { title } = await getPr(prId);
+
+  const body = {
+    title,
+    reviewers: reviewers.map(username => ({ username })),
+  };
```
Comment (rarkins) [inline lib/platform/bitbucket/index.js]: I'm a little surprised at this `body` definition. Does this mean that it's mandatory to include the `title` with this update, yet it's not mandatory to use the PR description as well?

File: lib/platform/bitbucket/index.js [hunk 2]
```diff
@@ -467,7 +476,11 @@ async function createPr(
     `/2.0/repositories/${config.repository}/pullrequests`,
     { body }
   )).body;
-  const pr = { id: prInfo.id, displayNumber: `Pull Request #${prInfo.id}` };
+  const pr = {
+    id: prInfo.id,
+    number: prInfo.id,
```
Comment (rarkins) [inline lib/platform/bitbucket/index.js]: Was this addressing an unrelated bug, or something else?

Comment (mdlavin) [reply lib/platform/bitbucket/index.js]: This was addressing a bug that blocked the implementation of `addReviewers`. In [the caller of addReviewers](https://github.com/renovatebot/renovate/blob/9b9604ac1670fc31f30abdfdfbfe1b468b7f5e50/lib/workers/pr/index.js#L364) the pr.number is used. Without this fix, the first argument to addReviewers for Bitbucket was `undefined`

File: lib/platform/bitbucket/index.js (hunk 1 above)
Comment (mdlavin) [reply lib/platform/bitbucket/index.js]: The requirement of a title and nothing else was surprising to me too. Without the title, Bitbucket will respond with a 400 status code and the following body content:

```json
{
  "type": "error",
  "error": {
    "fields": {"title": ["This field is required."]},
    "message": "title: This field is required."
  }
}
```

Comment (rarkins) [reply lib/platform/bitbucket/index.js]: Thanks for confirming. Can you also confirm that this approach doesn't accidentally *delete* the existing PR description?

File: lib/platform/bitbucket/index.js (hunk 2 above)
Comment (rarkins) [reply lib/platform/bitbucket/index.js]: Sounds like we should be returning only pr.number and not pr.id ! I'll look into this.

File: lib/platform/bitbucket/index.js (hunk 1 above)
Comment (mdlavin) [reply lib/platform/bitbucket/index.js]: It does not delete the existing PR description. I had the same concern, but when I tested against our Bitbucket repos it created PRs with good titles and descriptions. No data was lost when adding the reviewers

**Pull request merged**